string device_name
uint64 baud_rate
float32 protocol_version